lib/libuutil/uu_dprintf.c

```diff
@@ -76,14 +76,10 @@ uu_dprintf_create(const char *name, uu_dprintf_severity_t severity,
 	if ((D = uu_zalloc(sizeof (uu_dprintf_t))) == NULL)
 		return (NULL);
 
-	if (name != NULL) {
-		D->uud_name = strdup(name);
-		if (D->uud_name == NULL) {
-			uu_free(D);
-			return (NULL);
-		}
-	} else {
-		D->uud_name = NULL;
+	D->uud_name = strdup(name);
+	if (D->uud_name == NULL) {
+		uu_free(D);
+		return (NULL);
```

Comment:
The intention of this function is clearly to allow name to be NULL.  Rather than removing this code I'd recommend only running the `uu_check_name()` when `name != NULL`.  I'd suggest moving to it's own PR too so we can merge the other parts of this PR.